trusted-types/eval-function-constructor.html

```diff
@@ -32,15 +45,27 @@
   for (let mask = 0; mask < arg_max; mask++) {
     test(t => {
       assert_throws_js(EvalError,
-                       _ => new_function_with_maybe_trusted_args(mask));
+                       _ => new_function_with_maybe_trusted_args(mask, "Function"));
     }, "Function constructor with mixed plain and trusted strings, mask #" + mask);
+    test(t => {
+      assert_throws_js(EvalError,
+                       _ => new_function_with_maybe_trusted_args(mask, "AsyncFunction"));
+    }, "AsyncFunction constructor with mixed plain and trusted strings, mask #" + mask);
+    test(t => {
+      assert_throws_js(EvalError,
+                       _ => new_function_with_maybe_trusted_args(mask, "GeneratorFunction"));
+    }, "GeneratorFunction constructor with mixed plain and trusted strings, mask #" + mask);
+    test(t => {
+      assert_throws_js(EvalError,
+                       _ => new_function_with_maybe_trusted_args(mask, "AsyncGeneratorFunction"));
+    }, "AsyncGeneratorFunction constructor with mixed plain and trusted strings, mask #" + mask);
   }
 
   // Now do one with all trusted arguments. In that case, isTrusted=true in
   // EnsureCSPDoesNotBlockStringCompilation and the function is built without
   // error.
   test(t => {
-    const f = new_function_with_maybe_trusted_args(arg_max);
+    const f = new_function_with_maybe_trusted_args(arg_max, "Function");
```

Comment:
I guess the patch is ok like this. If you really want to be exhaustive, you could do:

```
["Function", "AsyncFunction", "GeneratorFunction", "GeneratorFunction", "AsyncGeneratorFunction"].forEach(functionConstructorType) => {

...
 for (let mask = 0; mask < arg_max; mask++) {
    test(t => {
...
    }, `${functionConstructorType} constructor with mixed plain and trusted strings, mask #${mask}`);
...
     test(t => {
...
      }, `${functionConstructorType} constructor with mixed plain and trusted strings, mask #${arg_max}`);
...
 test(t => {
...
    }, `${functionConstructorType} constructor with trusted strings, and a forged toString() for the one at index ${index}`);
});
```

ensure is `new Function` is replaced with the proper call depending on `functionConstructorType`.